src/ProjectService.Business/Commands/Project/EditProjectCommand.cs

```diff
@@ -20,123 +21,82 @@
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
 
-            return null;
-        }
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.AddRange(errors);
 
-        public EditProjectCommand(
-            IEditProjectValidator validator,
-            IAccessValidator accessValidator,
-            IPatchDbProjectMapper mapper,
-            IUserRepository userRepository,
-            IProjectRepository projectRepository,
-            IRequestClient<IGetDepartmentsRequest> rcGetDepartments,
-            IHttpContextAccessor httpContextAccessor,
-            ILogger<CreateProjectCommand> logger)
-        {
-            _validator = validator;
-            _accessValidator = accessValidator;
-            _mapper = mapper;
-            _projectRepository = projectRepository;
-            _userRepository = userRepository;
-            _rcGetDepartments = rcGetDepartments;
-            _httpContextAccessor = httpContextAccessor;
-            _logger = logger;
-        }
+        return response;
+      }
 
-        public OperationResultResponse<bool> Execute(Guid projectId, JsonPatchDocument<EditProjectRequest> request)
-        {
-            _validator.ValidateAndThrowCustom(request);
+      Guid userId = _httpContextAccessor.HttpContext.GetUserId();
 
-            DbProject dbProject = _projectRepository.Get(new GetProjectFilter { ProjectId = projectId });
+      if (!_accessValidator.HasRights(Rights.AddEditRemoveProjects)
+        && !_userRepository.AreUserProjectExist(userId, projectId, true))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
 
-            OperationResultResponse<bool> response = new();
-            Guid userId = _httpContextAccessor.HttpContext.GetUserId();
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.Add("Not enough rights.");
 
-            if (!_accessValidator.IsAdmin() &&
-                !_userRepository.AreUserProjectExist(userId, projectId, true) &&
-                GetDepartment(dbProject.DepartmentId, response.Errors)?.DirectorUserId != userId)
-            {
-                throw new ForbiddenException("Not enough rights.");
-            }
+        return response;
+      }
 
-            foreach (Operation item in request.Operations)
-            {
-                if (item.path == $"/{nameof(EditProjectRequest.Name)}" &&
-                    _projectRepository.IsProjectNameExist(item.value.ToString()))
-                {
-                    response.Status = OperationResultStatusType.Failed;
-                    response.Errors.Add($"Project with name '{item.value}' already exist");
-                    return response;
-                }
+      DbProject dbProject = _projectRepository.Get(new GetProjectFilter { ProjectId = projectId });
+      response.Body = _projectRepository.Edit(dbProject, _mapper.Map(request));
```

Comment:
For what do you pull dbProject in command? You can get dbProject in repo in Edit method

Reply:
done